packages/room/peer/peer.js

```diff
@@ -451,171 +425,210 @@ export const createPeer = ({ api, createStream, event, streams, config }) => {
+        window.RTCRtpTransceiver &&
+        'setCodecPreferences' in window.RTCRtpTransceiver.prototype
+      const browserName = getBrowserName() || ''
       /** @type {MediaStreamTrack | undefined} */
       const videoTrack = stream.mediaStream.getVideoTracks()[0]
+      /** @type {RTCRtpCodecCapability[]} */
+      const preferredCodecs = []
+
+      if (videoTrack) {
+        const systemVideoCodecs =
+          RTCRtpSender.getCapabilities('video')?.codecs || []
+
+        for (const videoCodec of config.media[type].videoCodecs) {
+          for (const systemVideoCodec of systemVideoCodecs) {
+            if (
+              systemVideoCodec.mimeType.toLowerCase() ===
+              videoCodec.toLowerCase()
+            ) {
+              preferredCodecs.push(systemVideoCodec)
+              systemVideoCodecs.splice(
+                systemVideoCodecs.indexOf(systemVideoCodec),
+                1
+              )
+            }
+          }
+        }
 
-      const browserName = getBrowserName()
-      const simulcastBrowsers = [SAFARI, CHROME, EDGE, OPERA]
+        // Add all remaining codecs
+        for (const videoCodec of systemVideoCodecs) {
+          preferredCodecs.push(videoCodec)
+        }
 
-      let svc = false
+        /** @type {RTCRtpTransceiverInit} */
+        const transceiverInit = {
+          direction: 'sendonly',
+          streams: [stream.mediaStream],
+        }
 
-      /** @type {RTCRtpCodecCapability[]} */
-      const videoPreferedCodecs = []
-      const videoCodecs = RTCRtpReceiver.getCapabilities('video')?.codecs
+        const svcEnabled =
+          config.media[type].svc &&
+          config.media[type].scalabilityMode &&
+          config.media[type].simulcast &&
+          config.media[type].maxFramerate &&
+          browserName !== FIREFOX
 
-      if (videoCodecs) {
-        if (this._videoCodecPreferences.length > 0) {
-          for (const videoCodecPreference of this._videoCodecPreferences) {
-            for (const videoCodec of videoCodecs) {
-              if (videoCodec.mimeType === videoCodecPreference) {
-                videoPreferedCodecs.push(videoCodec)
+        const simulcastEnabled =
+          config.media[type].simulcast && config.media[type].maxFramerate
 
-                if (
-                  videoCodec.mimeType === 'video/VP9' &&
-                  browserName !== FIREFOX &&
-                  stream.source !== 'screen'
-                ) {
-                  svc = true
-                }
-              }
-            }
+        if (svcEnabled) {
+          /** @type {import('../peer/peer-types.js').RoomPeerType.RTCRtpSVCEncodingParameters} */
+          const scalableEncoding = {
+            maxBitrate: this._highBitrate,
+            scalabilityMode: config.media[type].scalabilityMode,
+            maxFramerate: config.media[type].maxFramerate,
           }
+          transceiverInit.sendEncodings = [scalableEncoding]
+        } else if (simulcastEnabled) {
```

Comment:
The simulcast should not use else if but it separate conditional if. So it is possible that svc and simulcast used together but not recommended for low power CPU. Usually when simulcast and svc used together, the svc will only enable temporal layers and not using spatsial layer. Check this blog post: https://www.meetecho.com/blog/vp9-av1-simulcast-svc/

Reply:
Mas @tyohan I think I misunderstood about simulcast and svc. I thought svc can only be used when simulcast is enabled. However, those two are different technologies and can be used and activated together or separately. 

I will update the code.